### ISSUE TITLE
Codox with ClojureScript produces numerous warnings

### ISSUE THREAD

**Issue by metametadata** [weavejester/codox#123]
*Steps:*

1) Clone repo, cd into example/.
2) `lein with-profile +cljs codox`

*Expected:*
Docs generated with no warnings.

*Actual:*

```
WARNING: Use of undeclared Var cljs.core/not at line 43 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/not at line 43 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/not at line 43 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/not at line 43 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/not at line 43 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/missing-protocol at line 43 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/not at line 43 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/not at line 43 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/not at line 43 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/not at line 43 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/not at line 43 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/missing-protocol at line 43 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/atom at line 48 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/atom at line 48 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/atom at line 48 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/atom at line 48 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/get at line 48 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/get-global-hierarchy at line 48 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/MultiFn at line 48 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/MultiFn at line 48 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/symbol at line 48 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Bad method signature in protocol implementation, cljs.core/IWithMeta does not declare method called -with-meta at line 66 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Bad method signature in protocol implementation, cljs.core/IMeta does not declare method called -meta at line 66 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Bad method signature in protocol implementation, IDeref does not declare method called -deref at line 66 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/IWithMeta at line 66 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Symbol cljs.core/IWithMeta is not a protocol at line 66 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/IMeta at line 66 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Symbol cljs.core/IMeta is not a protocol at line 66 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var codox.example3/IDeref at line 66 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Symbol IDeref is not a protocol at line 66 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
WARNING: Use of undeclared Var cljs.core/-write at line 66 /Users/yuri/Dropbox/dev/codox/example/src/clojure/codox/example3.cljs
Generated HTML docs in /Users/yuri/Dropbox/dev/codox/example/target/base+system+user+dev+cljs/doc
```

I have the similar problem when generating docs for cljs code in my project.

**Comment by weavejester** [comment]
This may be an issue with the ClojureScript analyzer, but it would be nice to know if there's a way around this or not. This hasn't been a high priority issue for me because it doesn't affect the output at all, it just makes the doc generation unnecessarily noisy.

**Comment by metametadata** [comment]
Thanks for quick feedback! Indeed it looks like warnings don't affect doc output.

I've also tried [Codeína](https://github.com/funcool/codeina) fork and it doesn't emit warnings. My wild guess is that it's because of [using `an/no-warn`](https://github.com/funcool/codeina/blob/bd2ea93ef221c7b3022b1095f55584c0972af53e/src/codeina/reader/clojurescript.clj#L81).

**Comment by weavejester** [comment]
That might be a reasonable workaround. I'd accept a patch that adds that to Codox.

**Comment by metametadata** [comment]
I'm on it.

**Comment by weavejester** [comment]
Fixed by #124.
